test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -1425,15 +1469,18 @@ def _validate_argument_spec(self, docs, spec, kwargs, context=None):
                 file_common_arguments.update(data.get('aliases', []))
 
             args_from_docs = set()
-            for arg, data in docs.get('options', {}).items():
+            for arg, data in doc_options.items():
                 args_from_docs.add(arg)
                 args_from_docs.update(data.get('aliases', []))
 
+            # add_file_common_args is only of interest on top-level
+            add_file_common_args = kwargs.get('add_file_common_args', False) and not context
```

Comment:
What's the purpose of the `and not context` check here?

Reply:
It's to avoid that suboptions, which happen to be named as one of the `FILE_COMMON_ARGUMENTS` options, are ignored when they have no documentation, or that suboptions named like them are documented but don't actually exist.